packages/iTwinUI-react/src/core/Header/HeaderBasicButton.tsx

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { PolymorphicForwardRefComponent, useTheme } from '../utils';
+import { ButtonProps } from '../Buttons';
+
+export type HeaderBasicButtonProps = ButtonProps;
+
+type HeaderBasicButtonComponent = PolymorphicForwardRefComponent<
+  'button',
+  HeaderBasicButtonProps
+>;
+
+export const HeaderBasicButton: HeaderBasicButtonComponent = React.forwardRef(
+  (props, ref) => {
+    const {
+      onClick,
+      className,
+      children,
+      style,
+      type = 'button',
+      startIcon,
+      as: Element = 'button',
+      ...rest
+    } = props;
+
+    useTheme();
+
+    return (
+      <Element
+        className={cx('iui-header-breadcrumb-button', className)}
+        onClick={onClick}
+        ref={ref}
+        type={type}
+        style={style}
+        {...rest}
+      >
+        {startIcon &&
+          React.cloneElement(startIcon, {
+            className: startIcon.props.className,
+          })}
+        {children}
```

Comment:
wont we have breaking change now that we do not support `endIcon` ?

Reply:
`endIcon` support has been added to `HeaderBasicButton`. The `size` and `styleType` props are not needed for any of the `HeaderButton` types and were not used in the previous `HeaderButton`, so this should cause no breaking changes

Reply:
Its not about being used.. Its about the fact, that users could have passed those props.. 
So now if they pass, you spread them over the element (which is invalid)